e2e/LongHaul/service/IotHub.cs

```diff
@@ -20,29 +19,23 @@ internal class IotHub : IDisposable
         private readonly Logger _logger;
         private readonly string _hubConnectionString;
         private readonly IotHubTransportProtocol _transportProtocol;
-        private readonly string _deviceId;
         private readonly string _storageConnectionString;
 
-        private static readonly TimeSpan s_directMethodInvokeInterval = TimeSpan.FromSeconds(3);
-        private static readonly TimeSpan s_desiredPropertiesSetInterval = TimeSpan.FromSeconds(3);
-        private static readonly TimeSpan s_c2dMessagesSentInterval = TimeSpan.FromSeconds(3);
         private static readonly TimeSpan s_deviceCountMonitorInterval = TimeSpan.FromSeconds(30);
+        private static readonly TimeSpan s_receiveFileUploadInterval = TimeSpan.FromSeconds(30);
         private int _totalFileUploadNotificationsReceived;
 
         private static IotHubServiceClient s_serviceClient;
-        private static HashSet<string> s_activeDeviceSet;
         private BlobContainerClient _blobContainerClient;
 
-        private long _totalMethodCallsCount = 0;
-        private long _totalDesiredPropertiesUpdatesCount = 0;
-        private long _totalC2dMessagesSentCount = 0;
+        private static volatile Dictionary<string, Tuple<Action, CancellationTokenSource>> s_onlineDeviceOperations;
```

Comment:
I'm looking at only the code and I might not have the entire context of your change, so take this with a pinch of salt - is there a reason you're saving off an Action (and thus no return value) vs a Func (that returns a task). Since there is a cancellation token associated, Iwas expecting a Task in return that would signal how the Task ended.

Reply:
Yeah, that would make more sense.